### PR TITLE
Change the 2100 autostart to suitable the Bigfat plane.

### PR DESCRIPTION
**Pull request by CornerOfSkyline**
Change the 2100 autostart to suitable the Bigfat plane.

But the startup can not start the pwm out 5-7.

Please help me to find out where is wrong. Thank you!

### CODE REVIEW

File: src/modules/systemlib/mixer/mixer_load.c [hunk 1]
```diff
@@ -106,8 +107,10 @@ int load_mixer_file(const char *fname, char *buf, unsigned maxlen)
 		}
 
 		/* add the line to the buffer */
-		strcat(buf, line);
-	}
+        strcat(buf, line);
```
Comment (LorenzMeier) [inline src/modules/systemlib/mixer/mixer_load.c]: Why did you modify this? This doesn't look good.

File: ROMFS/px4fmu_common/mixers/BigFat.main.mix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+S: 0 3      0  20000 -10000 -10000  10000
+
+Reserved
+---------------
+Z:
+
+Aileron 1 mixer(Left)
+---------------
+M: 1
+O:       10000  10000      	0 -10000  10000
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/mixers/BigFat.main.mix]: The indentation here looks suspicious.

File: ROMFS/px4fmu_common/mixers/BigFat.aux.mix [hunk 3]
```diff
@@ -0,0 +1,19 @@
+V-shape Vertical Tail 1 mixer(Right)
+--------------
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/mixers/BigFat.aux.mix]: This makes no sense. Why do you have this file?

File: ROMFS/px4fmu_common/init.d/2100_mpx_easystar [hunk 4]
```diff
@@ -9,4 +9,17 @@
...
  
-set MIXER easystar
+set MIXER BigFat
+# PX4IO PWM output will be used by default
+set PWM_OUT 1    # Change parameters for the first 4 outputs
+#set PWM_RATE 400        # Set PWM rate to 400 Hz for better performance
+set PWM_DISARMED 900    # Motors should stop at this PWM value
+set PWM_MIN 1000
+set PWM_MAX 2000
+set FAILSAFE 1500
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/init.d/2100_mpx_easystar]: I don't think there is a FAILSAFE environment variable. This is wrong.

File: CMakeLists.txt.user.3.2-pre1 [hunk 5]
```diff
@@ -0,0 +1,234 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (LorenzMeier) [inline CMakeLists.txt.user.3.2-pre1]: Don't commit user files!

File: src/modules/systemlib/mixer/mixer_load.c (hunk 1 above)
Comment (CornerOfSkyline) [inline src/modules/systemlib/mixer/mixer_load.c]: I just want to see the mixer file is load correctly through the nsh.

File: ROMFS/px4fmu_common/mixers/BigFat.main.mix (hunk 2 above)
Comment (CornerOfSkyline) [inline ROMFS/px4fmu_common/mixers/BigFat.main.mix]: Did this file have something wrong?   I checked and  revised this file many times. I still don't know why it is not work.

File: ROMFS/px4fmu_common/mixers/BigFat.aux.mix (hunk 3 above)
Comment (CornerOfSkyline) [inline ROMFS/px4fmu_common/mixers/BigFat.aux.mix]: The main output didn't work fine with the main.mix, so i want to use the aux output to control the servos.

File: ROMFS/px4fmu_common/init.d/2100_mpx_easystar (hunk 4 above)
Comment (CornerOfSkyline) [inline ROMFS/px4fmu_common/init.d/2100_mpx_easystar]: Where set the FAILSAFE parm ? I see it is used in the rc.interface file.

File: CMakeLists.txt.user.3.2-pre1 (hunk 5 above)
Comment (CornerOfSkyline) [inline CMakeLists.txt.user.3.2-pre1]: OK.

File: src/modules/systemlib/mixer/mixer_load.c (hunk 1 above)
Comment (LorenzMeier) [inline src/modules/systemlib/mixer/mixer_load.c]: You modified the program flow while doing that.

**Comment by LorenzMeier** [comment]
I would recommend you to check out master fresh, re-do your mixer and try this again with minimal changes. Some of the things you did could be easily enough to let the mixer load fail, in particular the change to the mixer load logic.

**Comment by CornerOfSkyline** [comment]
OK. I will try again.

**Pull request closed without merging**